prediction/src/algorithms/segment/trained_model.py

```diff
@@ -36,15 +37,31 @@ def predict(dicom_path, centroids):
         dict: Dictionary containing path to serialized binary masks and
             volumes per centroid with form::
             {'binary_mask_path': str,
-             'volumes': list[float]}
+             'volumes': list[float]
+             'diameters': list[float]}
     """
-    load_ct(dicom_path)
+
+    reader = sitk.ImageSeriesReader()
+    reader.SetFileNames(reader.GetGDCMSeriesFileNames(dicom_path))
+    image = reader.Execute()
+
+    binary_mask = np.zeros_like(sitk.GetArrayFromImage(image), dtype=bool)
+
+    volumes = []
+    diameters = []
+    for nodule in centroids:
+        mask, diameter, volume = levelset_segment.segment_nodule(image, nodule)
+        volumes.append(volume)
+        diameters.append(diameter)
+        binary_mask |= sitk.GetArrayFromImage(mask).astype(bool)
+
     segment_path = os.path.join(os.path.dirname(__file__),
```

Comment:
Out of interest, don't we have some kind of ``settings.TEST_DATA`` directory setting? :)

Reply:
Not to my knowledge? But this is not test data at all - this is the binary mask produced by the algorithm. Ideally, we would use a temporary file, but someone has to clean it up after use, so I thought it would be better to just reuse the same filename.

Reply:
What's wrong with a tempfile? Specifically, `tempfile.NamedTemporaryFile`

> This function operates exactly as TemporaryFile() does, except that the file is guaranteed to have a visible name in the file system (on Unix, the directory entry is not unlinked). That name can be retrieved from the name attribute of the returned file-like object. Whether the name can be used to open the file a second time, while the named temporary file is still open, varies across platforms (it can be so used on Unix; it cannot on Windows NT or later). If delete is true (the default), the file is deleted as soon as it is closed.

Reply:
The only problem with tempfile is who has the responsibility of removing it? The client would then have to do the cleanup, wouldn't it?
I can certainly implement this, but I just didn't want us ending up in a situation where we have a harddrive filled with random binary masks.

Reply:
Well, more likely we would close the file after writing, which would delete it before the client could load it.

Reply:
Well, more likely we would close the file after writing, which would delete it before the client could load it.

Reply:
I've implemented something similar here:

https://anonscm.debian.org/git/reproducible/diffoscope.git/tree/diffoscope/tempfiles.py#n29

.. notice how it adds files to a global which is called when we exit.

Reply:
Ok, I've done something similar, and added cleanup via atexit. Still means we will have to restart the service occasionally, but I guess that shouldn't be a problem.